### PR TITLE
feat: add endpoints to oauth2 provider apps response

### PR DESCRIPTION
**Pull request by code-asher**
These will show up when configuring the application along with the client ID and everything else.  Should make it easier to use the application, otherwise you will have to go look up the URLs in the docs (which are not yet written).

Mostly extracted from https://github.com/coder/coder/pull/11609 to make that PR smaller, except I added a list to display them in the frontend as well.

### CODE REVIEW

File: site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPageView.tsx [hunk 1]
```diff
@@ -139,16 +139,28 @@ export const EditOAuth2AppPageView: FC<EditOAuth2AppProps> = ({
...
-            <CopyableValue value={app.id}>
-              {app.id}{" "}
-              <CopyIcon
-                css={{
-                  width: 16,
-                  height: 16,
-                }}
-              />
-            </CopyableValue>
+            <dl css={styles.dataList}>
```
Comment (Parkreiner) [inline site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPageView.tsx:142]: Big fan of the definition list usage

**Review by buenos-nachos — COMMENTED** [review]
Checked the stories, and the client-side stuff looks good! If there aren't any issues with the Go code, the PR should be good to approve

**Review by Emyrk — COMMENTED** [review]
Can you screencap what this looks like?

**Review by Emyrk — APPROVED** [review]
The Go looks fine though :+1:

**Comment by code-asher** [comment]
> Can you screencap what this looks like?

It can be viewed and messed around with here on Chromatic: https://624de63c6aacee003aa84340-yiqrjtuxwr.chromatic.com/?path=/story/pages-deploysettingspage-editoauth2apppageview--default

**Pull request merged**